larpandoracontent/LArThreeDReco/LArTrackFragments/ClearTrackFragmentsTool.cc

```diff
@@ -367,30 +360,6 @@ void ClearTrackFragmentsTool::RebuildClusters(ThreeDTrackFragmentsAlgorithm *con
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
-void ClearTrackFragmentsTool::UpdateTensor(ThreeDTrackFragmentsAlgorithm *const pAlgorithm, const TensorType &overlapTensor,
-    const ClusterList &clustersToRemoveFromTensor, const ClusterList &clustersToAddToTensor) const
-{
-    for (ClusterList::const_iterator iter = clustersToRemoveFromTensor.begin(), iterEnd = clustersToRemoveFromTensor.end(); iter != iterEnd; ++iter)
-        pAlgorithm->UpdateUponDeletion(*iter);
-
-    ClusterList newKeyClusters;
-    pAlgorithm->SelectInputClusters(&clustersToAddToTensor, newKeyClusters);
-
-    for (ClusterList::const_iterator iter = newKeyClusters.begin(), iterEnd = newKeyClusters.end(); iter != iterEnd; ++iter)
-        pAlgorithm->UpdateForNewCluster(*iter);
-
-    ClusterList affectedKeyClusters;
-    this->GetAffectedKeyClusters(overlapTensor, clustersToRemoveFromTensor, affectedKeyClusters);
-
-    for (ClusterList::const_iterator iter = affectedKeyClusters.begin(), iterEnd = affectedKeyClusters.end(); iter != iterEnd; ++iter)
-        pAlgorithm->UpdateUponDeletion(*iter);
-
-    for (ClusterList::const_iterator iter = affectedKeyClusters.begin(), iterEnd = affectedKeyClusters.end(); iter != iterEnd; ++iter)
-        pAlgorithm->UpdateForNewCluster(*iter);
-}
-
-//------------------------------------------------------------------------------------------------------------------------------------------
-
```

Comment:
I agree with removing this, its clearer by adding the relevant operations inside ProcessTensorElement